staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -545,6 +550,20 @@ func validateConversionReviewVersions(versions []string, requireRecognizedVersio
 	return allErrs
 }
 
+// Allows invalid CA Bundle to be specified only if the existing CABundle is invalid
+// or if the CRD is not established yet.
+func allowInvalidCABundle(oldCRD *apiextensions.CustomResourceDefinition) bool {
+	if !apiextensions.IsCRDConditionTrue(oldCRD, apiextensions.Established) {
```

Comment:
in the controller where we set Established=true (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/establish/establishing_controller.go#L119), after names are accepted, we now need to check if the caBundle is valid before setting Established=True

if it is not, set Established=False with a specific reason indicating the caBundle is not valid

Reply:
Updated and added tests in `[staging/src/k8s.io/apiextensions-apiserver/test/integration/cabundle_test.go](https://github.com/kubernetes/kubernetes/pull/124061/files#diff-e37b6fd927b1731dea94f57af2c0147367c3face7f0dca0f09c456b7117a539d)`